### ISSUE TITLE
Android-Scan-Unable to create split scan request with 2 participants, unexpected error displayed

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#31763]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.2-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Launch app
2. Tap request---scan
3. Choose a image from gallary
4. Select 2 participants to split
5. Tap add to split
6. Tap split
8. Open Split scan request displayed
9. Enter all details
10. Tap split and wait for few seconds

## Expected Result:
User must be able to create split scan request with 2 participants

## Actual Result:
When user create split scan request with 2 participants, error message "Unexpected error requesting money, please try again " is displayed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/d538f092-671a-4727-92f9-18deeaeb3c35

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by blimpich** [comment]
working on this, trying to find culprit PR

**Comment by blimpich** [comment]
None of the PRs particularly stand out to me. Put out [a feeler on #expensify-open-source slack channel](https://expensify.slack.com/archives/C01GTK53T8Q/p1700699287133899) to hopefully get more eyes on this. I'm currently attempting to repro this on a local Android virtual device, but having difficulty.

**Comment by situchan** [comment]
I have been researching on this. Will update in a few min

**Comment by luacmartins** [comment]
@situchan can you reproduce the issue?

**Comment by situchan** [comment]
yes, I narrowed PRs list. Found commit hashes between repro and not repro

**Comment by luacmartins** [comment]
do you have a commit hash that caused the issue?

**Comment by situchan** [comment]
I finally found offending PR - https://github.com/Expensify/App/pull/28449. As it's TS migration, better to revert.
Here's revert PR - https://github.com/Expensify/App/pull/31767 - just tested and not able to repro

**Comment by luacmartins** [comment]
Thanks! Can you elaborate why that PR caused the issue?

**Comment by situchan** [comment]
Just found the offending line. Commented on PR.

**Comment by situchan** [comment]
As we found fix, no need to revert. I just tested again with this fix.

**Comment by situchan** [comment]
So this is dupe of #26940. Fixed in #28533 but migration PR reverted that fix

**Comment by luacmartins** [comment]
Wanna work on a fix PR? I'll assign you to the issue

**Comment by situchan** [comment]
Sure. On PR now

**Comment by luacmartins** [comment]
We Cped a fix for this removing the blocker label.

**Comment by JmillsExpensify** [comment]
Is any payment required on this issue, seeing as it's apparently a dupe?

**Comment by situchan** [comment]
Payment required here. The dupe issue was very old one.

**Comment by JmillsExpensify** [comment]
@situchan mind completing the checklist ahead of the regression period passing. Thank you!

**Comment by JmillsExpensify** [comment]
Great thank you! Payment summary:

 - Contributor+: @situchan $750 for PR

**Comment by JmillsExpensify** [comment]
@situchan I went ahead and sent you an offer via Upwork.
